File: PRACTICA#3/PERSISTENCIA/3/JAVA/Main.java
/*
 * 
 * ----------------- EJERCICIO 3 -----------------
 * 
 * 
a) Implementar el diagrama de clases.
b) Implementa buscarCliente(int c) a través del id.
c) Implementa buscarCelularCliente(int c), que devuelva los datos del cliente
junto al número de celular.
 * 
 * 
 */

public class Main {
    public static void main(String[] args) {
        ArchivoCliente archivo = new ArchivoCliente("d:/UMSA/SEGUNDO SEMESTRE/PROGRAMACION II/PRACTICAS AUX/PRACTICA#3/PERSISTENCIA/3/JAVA/clientes.txt");
        archivo.crearArchivo();
        
        // a)
        Cliente cliente1 = new Cliente(1, "Juan", 123456789);
        Cliente cliente2 = new Cliente(2, "Maria", 987654321);
        archivo.guardarCliente(cliente1);
        archivo.guardarCliente(cliente2);
        
        // b)
        Cliente encontrado = archivo.buscarCliente(2);
        if (encontrado != null) {
            System.out.println(encontrado);
        } else {
            System.out.println("Cliente no encontrado");
        }
        
        // c)
        Cliente celular = archivo.buscarCelularCliente(1);
        if (celular != null) {
            System.out.println("Cliente con celular: " + celular);
        } else {
            System.out.println("Cliente no encontrado");
        }
    }
}

File: PRACTICA#3/PERSISTENCIA/3/PYTHON/Main.py
'''

------------- EJERCICIO 3 -------------


a) Implementar el diagrama de clases.
b) Implementa buscarCliente(int c) a través del id.
c) Implementa buscarCelularCliente(int c), que devuelva los datos del cliente
junto al número de celular.
'''


from Cliente import Cliente
from ArchivoCliente import ArchivoCliente

def main():
    archivo = ArchivoCliente("d:/UMSA/SEGUNDO SEMESTRE/PROGRAMACION II/PRACTICAS AUX/PRACTICA#3/PERSISTENCIA/3/PYTHON/clientes.txt")
    archivo.crearArchivo()
    
    # a)
    cliente1 = Cliente(1, "Juan", 123456789)
    cliente2 = Cliente(2, "Maria", 987654321)
    archivo.guardarCliente(cliente1)
    archivo.guardarCliente(cliente2)
    
    # b)
    encontrado = archivo.buscarCliente(2)
    if encontrado:
        print(encontrado)
    else:
        print("Cliente no encontrado")
    
    # c)
    celular = archivo.buscarCelularCliente(1)
    if celular:
        print("Cliente con celular:", celular)
    else:
        print("Cliente no encontrado")

if __name__ == "__main__":
    main()